Sources/Transaction.swift

```diff
@@ -10,13 +10,40 @@ import BigInt
 public struct Transaction {
     public var accountNonce: UInt64
     public var price: BigInt
-    public var gasLimit: BigInt
-    public var recipient: Address?
+    public var gasLimit: UInt64
+    public var recipient: Address
     public var amount: BigInt
     public var payload: Data?
 
     // Signature values
-    public var v: BigInt
-    public var r: BigInt
-    public var s: BigInt
+    public var v = BigInt()
+    public var r = BigInt()
+    public var s = BigInt()
+
+    /// Creates a `Transaction` with default values and the given recipient address.
+    public init(recipient: Address) {
```

Comment:
did you consider leaving initializer for all the values for safety? to avoid any issues, in that case creator would be aware of all the values to pass in, otherwise transaction won't succeed if everything is 0

Reply:
ok